lib/utils.ts

```diff
@@ -54,6 +54,38 @@ export const formatTransferRestriction = (
   }
 };
 
+export const getSafeChainAbbreviation = (chain: Chain | undefined) => {
+  if (!chain) {
+    return "";
+  }
+  switch (chain.id) {
+    case 11155111:
+      return "sep";
+    case 8453:
+      return "base";
+    case 10:
+      return "oeth";
+    case 84532:
+      return "basesep";
+    case 42220:
+      return "celo";
+    case 42161:
+      return "arb1";
+    default:
+      return chain.name;
+  }
+};
+
+export const generateSafeAppLink = (
+  chain: Chain | undefined,
+  safeAddress: `0x${string}`,
+) => {
+  if (!chain) {
+    return "";
+  }
+  return `https://app.safe.global/transactions/queue?safe=${getSafeChainAbbreviation(chain)}:${safeAddress}`;
```

Comment:
This will not work on FVM and other networks with a custom deployment of the Safe app.